### PR TITLE
Include Reminder command 

### LINKED ISSUE #43: As a user I want to set reminders for internship deadlines/appointments
... so that I can make sure I do not miss any internship opportunities

### LINKED ISSUE #72: Implement Reminder feature
Update the default list view of the Internship Diary to be shown according to urgency of the application, in terms of the application deadline followed by interview(s) date(s).

### PR DESCRIPTION
**Pull request by FeliciaTay**
The following changes were made:

- Addition of Reminder command to list out applications with application dates or interviews dates due in 7 days (in order of urgency)

Do help me review the changes I made too so that this pull request can be merged.

### CODE REVIEW

File: src/main/java/seedu/address/model/InternshipDiary.java [hunk 1]
```diff
@@ -105,7 +106,7 @@ public String toString() {
 
     @Override
     public ObservableList<InternshipApplication> getInternshipList() {
-        return internships.asUnmodifiableObservableList();
+        return internships.asUnmodifiableObservableList().sorted(new ApplicationDateThenInterviewDateComparator());
```
Comment (gerhean) [inline src/main/java/seedu/address/model/InternshipDiary.java]: Why is internship list being sorted by default? This is causing test to fail.

File: src/test/java/seedu/address/model/internship/ApplicationDateDuePredicateTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        assertFalse(applicationDateDuePredicate.equals(null));
+    }
+
+    @Test
+    public void test_futureApplicationDateIsWithin7Days_returnsTrue() {
+        ApplicationDateDuePredicate predicate = new ApplicationDateDuePredicate();
+
+        // application date is same as current date
+        LocalDate currentDate = LocalDate.now();
+        ApplicationDate newDate = new ApplicationDate(currentDate.format(DateTimeFormatter.ofPattern("dd MM YYYY")));
```
Comment (gerhean) [inline src/test/java/seedu/address/model/internship/ApplicationDateDuePredicateTest.java]: Can pass in LocalDate directly

File: src/main/java/seedu/address/model/InternshipDiary.java (hunk 1 above)
Comment (FeliciaTay) [reply src/main/java/seedu/address/model/InternshipDiary.java]: Perhaps I heard wrongly the last meeting, I will remove it.

File: src/test/java/seedu/address/model/internship/ApplicationDateDuePredicateTest.java (hunk 2 above)
Comment (FeliciaTay) [reply src/test/java/seedu/address/model/internship/ApplicationDateDuePredicateTest.java]: ohh okay, thanks!

File: src/main/java/seedu/address/logic/comparator/ApplicationDateThenInterviewDateComparator.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            }
+            return dateOfInternship1.compareTo(dateOfInternship2);
+        }
+
+        return dateOfInternship1.compareTo(dateOfInternship2);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other instanceof ApplicationDateThenInterviewDateComparator;
```
Comment (gerhean) [inline src/main/java/seedu/address/logic/comparator/ApplicationDateThenInterviewDateComparator.java:41]: Maybe can check the todaydate is equal also

File: src/test/java/seedu/address/logic/comparator/ApplicationDateThenInterviewDateComparatorTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import seedu.address.model.internship.InternshipApplication;
+import seedu.address.model.internship.interview.Interview;
+import seedu.address.testutil.InternshipApplicationBuilder;
+
+class ApplicationDateThenInterviewDateComparatorTest {
+
+    @Test
+    public void compare() {
+        InternshipApplicationBuilder internship1 = new InternshipApplicationBuilder();
+        internship1.withApplicationDate(new ApplicationDate(LocalDate.now()));
```
Comment (gerhean) [inline src/test/java/seedu/address/logic/comparator/ApplicationDateThenInterviewDateComparatorTest.java]: Maybe a good idea to declare todaydate = localdate.now() at the start? Then tests wont fail at midnight lolol. (Also applies to other tests).

Comment (gerhean) [reply src/test/java/seedu/address/logic/comparator/ApplicationDateThenInterviewDateComparatorTest.java]: I personally dont like using LocalDate.now() too much cause its a static method.

Comment (FeliciaTay) [reply src/test/java/seedu/address/logic/comparator/ApplicationDateThenInterviewDateComparatorTest.java]: True! Thanks!

File: src/main/java/seedu/address/logic/comparator/ApplicationDateThenInterviewDateComparator.java (hunk 3 above)
Comment (FeliciaTay) [reply src/main/java/seedu/address/logic/comparator/ApplicationDateThenInterviewDateComparator.java:41]: Okay!

**Comment by gerhean** [comment]
Good job

**Pull request merged**